src/daft-connect/src/spark_analyzer/expr_resolver.rs

```diff
@@ -0,0 +1,227 @@
+use std::sync::Arc;
+
+use daft_core::datatypes::IntervalValue;
+use daft_dsl::{PlanRef, UnresolvedColumn};
+use spark_connect::{
+    expression::{
+        self as spark_expr,
+        cast::{CastToType, EvalMode},
+        literal::LiteralType,
+        sort_order::{NullOrdering, SortDirection},
+        Literal, UnresolvedFunction,
+    },
+    Expression,
+};
+use tracing::debug;
+
+use super::to_daft_datatype;
+use crate::{
+    error::{ConnectError, ConnectResult},
+    functions::CONNECT_FUNCTIONS,
+    invalid_argument_err, invalid_relation_err, not_yet_implemented,
+};
+
+#[derive(Debug, Clone)]
+pub(crate) struct ExprResolver {}
```

Comment:
Consider renaming `ExprResolver` and `ExprResolver::resolve_expr` to something else so that it doesn't conflate with our concept of expr resolution in the logical plan. Maybe `ExprTranslator` and `translate_expr`?